network/kademlia_test.go

```diff
@@ -462,6 +461,24 @@ func TestSuggestPeers(t *testing.T) {
 
 	//No more peers left in unsaturated bins
 	tk.checkSuggestPeer("<nil>", 0, false)
+}
+
+func TestSuggestPeersSaturationDepthChange(t *testing.T) {
+
+	base := "00000000"
+	tk := newTestKademlia(t, base)
+	tk.On("00100000", "00010000", "10000000", "11000000", "11100000", "01000000", "01100000")
+
+	//Saturation depth is 2
+	tk.Off("01000000")
+	//Saturation depth should have fallen to 1
+	tk.checkSuggestPeer("01000000", 1, true)
+	tk.On("01000000")
+
+	//Saturation depth is 2 again
+	tk.Off("10000000")
+	//Saturation depth should have fallen to 0
+	tk.checkSuggestPeer("10000000", 0, true)
```

Comment:
Please consider also messing with the neighbors, it involves a slightly different part of the calculation.